src/lib/utils/meta_tables/abstract_meta_system_table.cpp

```diff
@@ -0,0 +1,46 @@
+#ifdef __APPLE__
+#include <mach/mach.h>
+#endif
+
+#include <fstream>
+
+#include "utils/meta_tables/abstract_meta_system_table.hpp"
+
+namespace opossum {
+
+AbstractMetaSystemTable::AbstractMetaSystemTable(const TableColumnDefinitions& column_definitions)
+    : AbstractMetaTable(column_definitions) {}
+
+// Returns the number of logical processors
+size_t AbstractMetaSystemTable::_get_cpu_count() {
+#ifdef __linux__
+
+  std::ifstream cpu_info_file;
+  cpu_info_file.open("/proc/cpuinfo", std::ifstream::in);
+  DebugAssert(cpu_info_file.is_open(), "Failed to open /proc/cpuinfo");
+
+  size_t processors = 0;
+  std::string cpu_info_line;
+  while (std::getline(cpu_info_file, cpu_info_line)) {
+    if (cpu_info_line.rfind("processor", 0) == 0) ++processors;

```

Comment:
Minor: Not sure if both of our used libraries already support `string.starts_with`, but that would be slightly easier and more efficient.

Reply:
neat

Reply:
Forget about the performance thing though. I didn't know about the rfind trick. This is just syntactic sugar then.